crawler/file_processing.py

```diff
@@ -663,60 +717,394 @@ def format_and_filter_rows(self, csvreader: DictReader) -> Any:
+            if not row[FIELD_CH2_TARGET] in ALLOWED_CH_TARGET_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 17",
+                    f"CH2-Target invalid, line: {line_number}, result: {row[FIELD_CH2_TARGET]}",
+                )
+                return False
+
+        if FIELD_CH3_TARGET in row and row[FIELD_CH3_TARGET]:
+            if not row[FIELD_CH3_TARGET] in ALLOWED_CH_TARGET_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 17",
+                    f"CH3-Target invalid, line: {line_number}, result: {row[FIELD_CH3_TARGET]}",
+                )
+                return False
+
+        if FIELD_CH4_TARGET in row and row[FIELD_CH4_TARGET]:
+            if not row[FIELD_CH4_TARGET] in ALLOWED_CH_TARGET_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 17",
+                    f"CH4-Target invalid, line: {line_number}, result: {row[FIELD_CH4_TARGET]}",
+                )
+                return False
+
+        return True
+
+    def row_channel_result_values_valid(self, row, line_number) -> bool:
+        """Validation to check if the row channel result values match one of the expected values.
 
+        Arguments:
+            row {Row} - row object from the csvreader
+            line_number {integer} - line number within the file
+
+        Returns:
+            bool - whether the values are valid
+        """
+        if FIELD_CH1_RESULT in row and row[FIELD_CH1_RESULT]:
+            if not row[FIELD_CH1_RESULT] in ALLOWED_CH_RESULT_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 18",
+                    f"CH1-Result invalid, line: {line_number}, result: {row[FIELD_CH1_RESULT]}",
+                )
+                return False
+
+        if FIELD_CH2_RESULT in row and row[FIELD_CH2_RESULT]:
+            if not row[FIELD_CH2_RESULT] in ALLOWED_CH_RESULT_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 18",
+                    f"CH2-Result invalid, line: {line_number}, result: {row[FIELD_CH2_RESULT]}",
+                )
+                return False
+
+        if FIELD_CH3_RESULT in row and row[FIELD_CH3_RESULT]:
+            if not row[FIELD_CH3_RESULT] in ALLOWED_CH_RESULT_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 18",
+                    f"CH3-Result invalid, line: {line_number}, result: {row[FIELD_CH3_RESULT]}",
+                )
+                return False
+
+        if FIELD_CH4_RESULT in row and row[FIELD_CH4_RESULT]:
+            if not row[FIELD_CH4_RESULT] in ALLOWED_CH_RESULT_VALUES:
+                self.logging_collection.add_error(
+                    "TYPE 18",
+                    f"CH4-Result invalid, line: {line_number}, result: {row[FIELD_CH4_RESULT]}",
+                )
+                return False
+
+        return True
+
+
+    def is_number(self, item):
```

Comment:
Think you could slim this down by 
a) Get rid of isinstance() check - assume the input is a string, let it blow up if it's not (it will always be in our case right)
b) `return (decimal_pattern.match(item) or int_pattern.match(item))` instead of the if / else / return ?

Another option is to try / except casting to a Decimal